horaedb/metric_engine/src/storage.rs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use arrow::{array::RecordBatch, datatypes::Schema};
+use async_trait::async_trait;
+
+use crate::{
+    manifest::Manifest,
+    sst::SSTable,
+    types::{ObjectStoreRef, Predicate, SendableRecordBatchStream, TimeRange},
+    Result,
+};
+
+pub struct CompactContext {}
+pub struct WriteContext {}
+pub struct ScanContext {}
+
+/// Time-aware merge storage interface.
+#[async_trait]
+pub trait TimeMergeStorage {
```

Comment:
I think may be just define the `xxxRequest`? And we fill it when we impl it.

Reply:
Good idea.